Running-Mastodon/Production-guide.md

```diff
@@ -37,7 +37,9 @@ server {
   server_name example.com;
 
   ssl_protocols TLSv1.2;
-  ssl_ciphers HIGH:!MEDIUM:!LOW:!aNULL:!NULL:!SHA;
+  ssl_ciphers 'ECDHE+CHACHA20:ECDHE+AESGCM'; # or 'ECDHE:!DSS:!PSK:!SRP:!3DES:!RC4:!DES:!IDEA:!RC2:!NULL'
+  #ssl_dhparam ssl/dhparam.pem; # generate with “openssl dhparam -out dhparam.pem 4096”, only needed if you add DHE to the ciphers
+  ssl_ecdh_curve X25519:sect571r1:secp521r1:secp384r1:prime256v1; # If only one curve : prime256v1 (Android 7.0 compat) or secp384r1 (not compatible with Android 7.0)
```

Comment:
Ideal curves, but you can only specify multiple curves, or use `X25519`, `sect571r1` or `secp521r1 ` with OpenSSL 1.1.x+ or LibreSSL, which are not on Debian, Ubuntu etc, unless you use Debian testing or compile Nginx yourself.

With "classic" nginx + openssl installations, this will just throw errors (like we discussed on Mastodon...), so you can't put this in the docs. `secp384r1 ` is great but won't work with Tusky on Android 7.0.

Thus `ssl_ecdh_curve prime256v1;` seems to be the ideal curve (it's still safe and provides good compatibility)